testing/ef-tests/src/cases/blockchain_test.rs

```diff
@@ -45,6 +51,72 @@ pub struct BlockchainTestCase {
 }
 
+impl BlockchainTestCase {
+    /// Checks if the test case is a particular test called `UncleFromSideChain`
+    ///
+    /// This fixture fails as expected, however it fails at the wrong block number.
+    /// Given we no longer have uncle blocks, this test case was pulled out such
+    /// that we ensure it still fails, however we do not check the block number.
+    #[inline]
+    fn is_uncle_sidechain_case(name: &str) -> bool {
+        name.contains("UncleFromSideChain")
+    }
+
+    /// If the test expects an exception, return the the block number
+    /// at which it must occur together with the original message.
+    ///
+    /// Note: There is a +1 here because the genesis block is not included
+    /// in the set of blocks, so the first block is actual block number 1
+    /// and not block number 0.
+    #[inline]
+    fn expected_failure(case: &BlockchainTest) -> Option<(u64, String)> {
+        case.blocks.iter().enumerate().find_map(|(idx, blk)| {
+            blk.expect_exception.as_ref().map(|msg| ((idx + 1) as u64, msg.clone()))
+        })
+    }
+
+    /// Execute a single `BlockchainTest`, validating the outcome against the
+    /// expectations encoded in the JSON file.
+    fn run_single_case(name: &str, case: &BlockchainTest) -> Result<(), Error> {
+        let expectation = Self::expected_failure(case);
+        match run_case(case) {
+            // All blocks executed successfully.
+            Ok(()) => {
+                // Check if the test case specifies that it should have failed
+                if let Some((block, msg)) = expectation {
+                    Err(Error::Assertion(format!(
+                        "Test case: {name}\nExpected failure at block {block} - {msg}, but all blocks succeeded",
+                    )))
+                } else {
+                    Ok(())
+                }
+            }
+
+            // A block processing failure occurred.
+            Err(Error::BlockProcessingFailed { block_number }) => match expectation {
+                // It happened on exactly the block we were told to fail on
+                Some((expected, _)) if block_number == expected => Ok(()),
+
+                // Uncle side‑chain edge case, we accept as long as it failed.
+                // But we don't check the exact block number.
+                _ if Self::is_uncle_sidechain_case(name) => Ok(()),
+
+                // Expected failure, but block number does not match
+                Some((expected, _)) => Err(Error::Assertion(format!(
+                    "Test case: {name}\nExpected failure at block {expected}\nGot failure at block {block_number}",
+                ))),
+
+                // No failure expected at all - bubble up original error.
+                None => Err(Error::BlockProcessingFailed { block_number }),
+            },
+
+            // Non‑processing error – forward as‑is.
+            // TODO: Don't think these can occur anymore since we don't batch execute anymore.
```

Comment:
This can only happen if the post state root matched but for some reason, the post state values in the test did not match with the ones in the database

Reply:
Ah, I was using a newer version of the specs

Reply:
Opened up #15809 to update the specs. If that passes, I can make the change here and rebase when merged